backend/app/services/device_service.py

```
@@ -61,14 +70,18 @@ async def set_device_online(
+            "executor_version": executor_version,
         }
         result = await cache_manager.set(key, data, expire=DEVICE_ONLINE_TTL)
         logger.info(f"[DeviceService] set_device_online: key={key}, result={result}")
         return result
 
     @staticmethod
     async def refresh_device_heartbeat(
-        user_id: int, device_id: str, running_task_ids: list[int] = None
+        user_id: int,
+        device_id: str,
+        running_task_ids: list[int] = None,
```

Ruff (0.14.14)

[warning] PEP 484 prohibits implicit `Optional`

Convert to `T | None`

(RUF013)